include/wil/registry_helpers.h

```diff
@@ -1367,6 +1370,27 @@ namespace reg
             return current_length;
         }
 
+        // failure returns zero
+        inline size_t resize_name(::std::wstring& name, size_t current_length, size_t new_length) WI_NOEXCEPT
+        {
+            // because std::wstring encodes the size within its object, we must resize whenever the 2 are not equal
+            // not just when the new_length is greater than the current_length -- which is correct for types that don't encode their size
+            if (new_length != current_length)
+            {
+                // resize_buffer takes size in bytes
+                if (FAILED(::wil::reg::reg_view_details::reg_value_type_info::resize_buffer(
+                        name, static_cast<DWORD>(new_length * sizeof(wchar_t)))))
+                {
+                    return 0;
+                }
+                return new_length;
+            }
+
+            // continue to use the existing buffer since the requested length is less than or equals to the current length
+            clear_name(name, current_length);
+            return current_length;
+        }

```

Comment:
```suggestion
#if defined(WIL_ENABLE_EXCEPTIONS) && defined(_STRING_)
        // failure returns zero
        inline size_t resize_name(::std::wstring& name, size_t current_length, size_t new_length) WI_NOEXCEPT
        {
            // because std::wstring encodes the size within its object, we must resize whenever the 2 are not equal
            // not just when the new_length is greater than the current_length -- which is correct for types that don't encode their size
            if (new_length != current_length)
            {
                // resize_buffer takes size in bytes
                if (FAILED(::wil::reg::reg_view_details::reg_value_type_info::resize_buffer(
                        name, static_cast<DWORD>(new_length * sizeof(wchar_t)))))
                {
                    return 0;
                }
                return new_length;
            }

            // continue to use the existing buffer since the requested length is less than or equals to the current length
            clear_name(name, current_length);
            return current_length;
        }
#endif
```
Or move it up to a pre-existing guard. This should fix your build failure